includes/components/class-admin.php

```diff
@@ -246,6 +250,15 @@ public function init_settings() {
 		}
 	}
 
+	/**
+	 * Clears cached extensions.
+	 */
+	public function flush_extensions_cache() {
+
+		// Delete cached extensions.
+		hivepress()->cache->delete_cache( 'all_extensions' );
```

Comment:
The "all_extensions" cache key is not related to themes, please add 2 separate callbacks for hooks /activate and switch theme, named like clear_extensions_cache, clear_themes_cache

The themes cache uses "themes" cache key https://github.com/hivepress/hivepress/blob/55d2a19f53cf5e30ede24c70f6089bc68940b0f3/includes/components/class-admin.php#L622